labonneboite/web/app.py

```diff
@@ -323,11 +323,17 @@ def log_extra_context():
     return extra_msg
 
 
+@app.errorhandler(AuthFailed)
 @app.errorhandler(AuthCanceled)
-def social_auth_canceled(error):
+@app.errorhandler(AuthStateMissing)
+@app.errorhandler(AuthException)
+def social_auth_error(error):
     """
     Handle the situation where a user clicks the `cancel` button on a third party auth provider website.
     """
+    if type(error).__name__ in ["AuthFailed", "AuthCanceled", "AuthStateMissing"]:
+        flash(u"Une erreur est survenue lors de votre connexion. Veuillez réessayer", 'error')
```

Comment:
Je propose plutôt : 

    if not isinstance(error, AuthCanceled):
        flash(u"Une erreur est survenue lors de votre connexion. Veuillez réessayer", 'error')
        logger.warn("PEAM error: %s", error)

Reply:
apparemment ça marche même avec un tuple:

`if not isinstance(error, (AuthCanceled, AuthFailed, AuthStateMissing)):`

https://stackoverflow.com/questions/10458093/how-to-compare-inheritance-with-several-classes

Reply:
euh typo de régis!!  :)  pas de  `not` lol....

`if isinstance(error, (AuthCanceled, AuthFailed, AuthStateMissing)):`